docs(borealis): initial borealis CLI docs DOC-456 (#918)
armory/docs: content/en/borealis/quick-start/borealis-integrate.md

---
title: Integrate Borealis & Automate Deployments
linktitle: Integrate & Automate
description: >
  Project Borealis and the CLI can be integrated with any of your tools and scripts that support invoking a CLI as part of its workflow. This gives you the ability to automatically deploy apps using Borealis as part of existing workflows.
exlcude_search: true
---

> ## Requirements

Reviewer: ```suggestion
## {{% heading "prereq" %}}
```

> Before you get started, ensure that the requirements listed on [Requirements]({{< ref "borealis-requirements" >}}) are met and that you can log in to the [Armory Cloud Console](https://console.cloud.armory.io/).

Reviewer: ```suggestion
Ensure you meet the requirements listed on [Requirements]({{< ref "borealis-requirements" >}}) and that you can log in to the [Armory Cloud Console](https://console.cloud.armory.io/).
```

## Create service account credentials

> Service account credentials are machine-to-machine client credentials that you use to authenticate the Borealis CLI with Armory's hosted cloud services when the CLI is used as part of an automated workflow. These are passed through the `clientID` and `clientSecret` parameters.

Reviewer: ```suggestion
Service account credentials are machine-to-machine client credentials that the Borealis CLI uses to authenticate with Armory's hosted cloud services when you use CLI as part of an automated workflow. These credentials are passed through the `clientID` and `clientSecret` parameters.
```

To create service account credentials, perform the following steps:

{{< include "aurora-borealis/borealis-client-creds.md" >}}

Armory recommends that you store these credentials in a secret engine that is supported by the tool you want to integrate Borealis with.

## GitHub Actions

Armory has a GitHub action available on the marketplace for integrating Project Borealis with your GitHub repo. For more information, see TBD.

## Jenkins

Jenkins supports invoking a CLI as a step in a pipeline. 

## Tekton

> Tekton supports invoking a CLI as a step in a pipeline.

Reviewer 2: Do we intent to provide minimal examples here, or link to upstream documentation?

Author: I'd like to provide the info a user would need from us and then point them to the upstream docs cause who knows if Jenkins or Tekton will change something. A step-by-step example, I'd like to put in a blog so that we don't have the obligation of keeping it up-to-date. 

With that said, I'd be worried about Tekton doing something drastic as its a younger project when compared to Jenkins.
